chemprop/cli/train.py

```diff
@@ -542,170 +531,220 @@ def main(args):
                 **featurization_kwargs,
             )
             train_data = all_data
-        else:
-            if multicomponent:
-                train_data, val_data, _ = split_multicomponent(all_data, args.split, **split_kwargs)
-            else:
-                train_data, val_data, _ = split_monocomponent(all_data, args.split, **split_kwargs)
-    else:
+            needs_val_data = False
+    elif args.separate_val_path is not None:
         raise ArgumentError(
-            argument=None, message="'val_path' must be specified if 'test_path' is provided!"
+            argument=None, message="'test_path' must be specified if 'val_path' is provided!"
         )  # TODO: In v1 this wasn't the case?
```

Comment:
This TODO might need to be broken out into a separate issue?

Reply:
Thanks! I made this issue to keep track of this TODO: https://github.com/chemprop/chemprop/issues/639.